lib/liquid.ex

```diff
@@ -1,6 +1,26 @@
 defmodule Liquid do
   use Application
 
+  @type t :: [
```

Comment:
These are all tag types, why should they be `Liquid` types?

Reply:
Because the parser is recursive. Almost every _block_ tag allows strings, variables or tags inside which are represented by `Liquid.t()`

Reply:
Sorry let me rephrase the question:
Why should those form a public Liquid type and where would it get referenced outside of the parser? I don't see a need for it here, perhaps it fits better as the return type of Parser.parse?

Reply:
`Parser.parse` returns `[String.t() | some_tag() | liquid_varible]` which is the type described in `Liquid` module, however, we need a place to _declare_ the type and I think that the perfect place is inside `Liquid` module because this is the abstraction of what can be processed by liquid. Maybe other place can be the `Template` module.